modal/_utils/async_utils.py

```diff
@@ -454,121 +453,30 @@ async def asyncnullcontext(*args, **kwargs):
 SEND_TYPE = typing.TypeVar("SEND_TYPE")
 
 
-class NestedAsyncCalls(Exception):
-    pass
-
-
-class Runner:
-    """Simplified backport of asyncio.Runner from Python 3.11
-
-    Like asyncio.run() but allows multiple calls to the same event loop
-    before teardown.
-
-    Difference from running new_event_loop().run_until_complete is that
-    this catches SIGINTs and propagates it as task cancellations rather
-    than raising KeyboardInterrupt inside of the event loop code.
-    """
-
-    # TODO: unify this with modal._container_entrypoint.UserCodeEventLoop
-    #       which does very similar things but has some additional SIGUSR1
-    #       logic
-
-    def __enter__(self) -> "Runner":
-        try:
-            asyncio.get_running_loop()
-        except RuntimeError:
-            pass  # no event loop - this is what we expect!
-        else:
-            raise NestedAsyncCalls()
-
-        self._loop = asyncio.new_event_loop()
-        return self
-
-    def __exit__(self, exc_type, exc_value, traceback):
-        self._loop.run_until_complete(self._loop.shutdown_asyncgens())
-        if sys.version_info[:2] >= (3, 9):
-            # Introduced in Python 3.9
-            self._loop.run_until_complete(self._loop.shutdown_default_executor())
-
-        self._loop.close()
-        return False
-
-    def run(self, coro: typing.Awaitable[T]) -> T:
-        is_main_thread = threading.current_thread() == threading.main_thread()
-        self._num_sigints = 0
-
-        coro_task = asyncio.ensure_future(coro, loop=self._loop)
-
-        async def wrapper_coro():
-            # this wrapper is needed since run_coroutine_threadsafe *only* accepts coroutines
-            return await coro_task
-
-        def _sigint_handler(signum, frame):
-            # cancel the task in order to have run_until_complete return soon and
-            # prevent a bunch of unwanted tracebacks when shutting down the
-            # event loop.
-
-            # this basically replicates the sigint handler installed by asyncio.run()
-            self._num_sigints += 1
-            if self._num_sigints == 1:
-                # first sigint is graceful
-                self._loop.call_soon_threadsafe(coro_task.cancel)
-                return
-
-            # this should normally not happen, but the second sigint would "hard kill" the event loop
-            # by raising KeyboardInterrupt inside of it
-            raise KeyboardInterrupt()
-
-        original_sigint_handler = None
+def run_async_gen(
```

Comment:
Just pulled this out of the `Runner` here and made it use the identical `Runner` implementation from synchronicity itself instead